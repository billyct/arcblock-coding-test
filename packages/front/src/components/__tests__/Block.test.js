import {render} from '@testing-library/react'

jest.mock('../../hooks/useBlock')
const useBlock = require('../../hooks/useBlock')

jest.mock('../Error', () => 'error-component')
jest.mock('../Loading', () => 'loading-component')
jest.mock('../BlockHash', () => 'block-hash-component')
jest.mock('../BlockSummary', () => 'block-summary-component')
jest.mock('../BlockTransactions', () => 'block-transactions-component')

const Block = require('../Block').default

afterAll(() => {
  jest.resetModules()
})

describe.each([
  [
    {
      error: {
        message: 'a error message'
      }
    }
  ],
  [
    {block: undefined}
  ],
  [
    {block: 'a block'}
  ]
])(`test Block Component with useBlock(%o)`, data => {

  beforeAll(() => {
    useBlock.__setMockData(data)
  })

  it('should render correct', function () {
    const {container} = render(<Block/>)
    expect(container).toMatchSnapshot()
  })
})
