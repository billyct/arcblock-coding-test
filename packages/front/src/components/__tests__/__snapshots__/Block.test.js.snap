// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Block Component with useBlock({ block: 'a block' }) should render correct 1`] = `
<div>
  <block-hash-component />
  <block-summary-component />
  <block-transactions-component />
</div>
`;

exports[`test Block Component with useBlock({ block: undefined }) should render correct 1`] = `
<div>
  <loading-component />
</div>
`;

exports[`test Block Component with useBlock({ error: { message: 'a error message' } }) should render correct 1`] = `
<div>
  <error-component>
    a error message
  </error-component>
</div>
`;
